plan/decorrelate.go

```diff
@@ -158,6 +160,11 @@ func (s *decorrelateSolver) optimize(p LogicalPlan, _ context.Context, _ *idAllo
 			}
 		}
 	}
+	if sel, ok := p.(*Selection); ok {
+		if ds, ok := p.(*DataSource); ok {
```

Comment:
p.GetChildren()[0]...
output controller to sel explain result. and add explain test for it.